doc/adr/0001-reactive-reassignment.md

```diff
@@ -0,0 +1,49 @@
+# 1. Reactive reassignment
+
+Date: 2018-11-21
+
+## Status
+
+Accepted
+
+## Context
+
+In current architecture, we receive always a batch of products. For this batch, we check if any products need reassignment.
+In order to find out, we search for conflicting products with query by SKUs from variants and slugs from all locale and both current and staged.
+This query can get too big for GET request and cannot be used.  
+
+## Decision
+
+Variant reassignment will be called only in case of errors and it will solve only the failed product.
+
+## Consequences
+Refactoring needs to be done both in reassignment module and also modules that uses reassignment (right now only sphere-product-import).
+
+#### sphere-product-import changes
+1. Catch errors and on errors with 400 HTTP status call variant reassignment with the failed product. Reassignment will run in these cases:
+    1. Error code is `DuplicateField` and field is `slug`
+    1. Error code is `DuplicateField` and field is `sku`
+    1. Error code is `InvalidOperation` and message contains `product type` 
+1. On product sync action generations check created actions. If one of the action is `removeVariant`, call reassignment module.
+Reassignment needs to backup this variant before deletion so we don't lose any data.
+1. After reassignment module is finished, sphere-product-import should again with the updated product.
+
+
+## Further improvements
+
+#### Variant reassignment changes
+1. Parse the error and check what is the cause of the error.
+1. If error is of type `DuplicateField`, process the error message and check if the error is concerning SKU or slugs.
+1. Variant reassignment will try to resolve only the errors that it receives. It will NOT proactively search for possible errors.
```

Comment:
only ~the~ errors

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/77/commits/6b966f6aee1249d5398b4b636b6c9803f78e56ef#diff-69de50b194fdf22d40e38b82f2045440R37